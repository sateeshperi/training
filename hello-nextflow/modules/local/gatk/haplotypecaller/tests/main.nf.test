nextflow_process {

    name "Test Process GATK_HAPLOTYPECALLER"
    script "../main.nf"
    process "GATK_HAPLOTYPECALLER"

    test("reads_son [bam]") {

        setup {
            run("SAMTOOLS_INDEX") {
                script "../../../samtools/index/main.nf"
                process {
                    """
                    input[0] =  [ [id: 'NA12882' ], file("/workspace/gitpod/hello-nextflow/data/bam/reads_son.bam") ]
                    """
                }
            }
        }

        when {
            params {
                outdir = "tests/results"
            }
            process {
                """
                input[0] = SAMTOOLS_INDEX.out
                input[1] = file("${baseDir}/data/ref/ref.fasta")
                input[2] = file("${baseDir}/data/ref/ref.fasta.fai")
                input[3] = file("${baseDir}/data/ref/ref.dict")
                input[4] = file("${baseDir}/data/intervals.list")
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }

    test("reads_mother [bam]") {

        setup {
            run("SAMTOOLS_INDEX") {
                script "../../../samtools/index/main.nf"
                process {
                    """
                    input[0] = [ [id: 'NA12878' ], file("/workspace/gitpod/hello-nextflow/data/bam/reads_mother.bam") ]
                    """
                }
            }
        }

        when {
            params {
                outdir = "tests/results"
            }
            process {
                """
                input[0] = SAMTOOLS_INDEX.out
                input[1] = file("${baseDir}/data/ref/ref.fasta")
                input[2] = file("${baseDir}/data/ref/ref.fasta.fai")
                input[3] = file("${baseDir}/data/ref/ref.dict")
                input[4] = file("${baseDir}/data/intervals.list")
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }

    test("reads_father [bam]") {

        setup {
            run("SAMTOOLS_INDEX") {
                script "../../../samtools/index/main.nf"
                process {
                    """
                    input[0] = [ [id: 'NA12877' ], file("/workspace/gitpod/hello-nextflow/data/bam/reads_father.bam") ]
                    """
                }
            }
        }

        when {
            params {
                outdir = "tests/results"
            }
            process {
                """
                input[0] = SAMTOOLS_INDEX.out
                input[1] = file("${baseDir}/data/ref/ref.fasta")
                input[2] = file("${baseDir}/data/ref/ref.fasta.fai")
                input[3] = file("${baseDir}/data/ref/ref.dict")
                input[4] = file("${baseDir}/data/intervals.list")
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }

}
