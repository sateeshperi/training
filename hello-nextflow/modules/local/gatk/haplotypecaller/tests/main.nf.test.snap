{
    "reads_father [bam]": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "NA12877"
                        },
                        "reads_father.bam.g.vcf:md5,dafb8478c3fe7fd436d152218f0435a5",
                        "reads_father.bam.g.vcf.idx:md5,b3ac7efdb22e113fa88c90f8a91e790a"
                    ]
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.02.0"
        },
        "timestamp": "2024-04-18T20:57:40.602376102"
    },
    "reads_son [bam]": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "NA12882"
                        },
                        "reads_son.bam.g.vcf:md5,96594b152dd3ed9a0a672bca305e1ae6",
                        "reads_son.bam.g.vcf.idx:md5,09bcb5615dfaab274d9632b689b2ed83"
                    ]
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.02.0"
        },
        "timestamp": "2024-04-18T20:57:02.95077265"
    },
    "reads_mother [bam]": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "NA12878"
                        },
                        "reads_mother.bam.g.vcf:md5,f386344bddd5c19bc87f6b32f2343bc3",
                        "reads_mother.bam.g.vcf.idx:md5,3c0992912535547bf29093c220645d1a"
                    ]
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.02.0"
        },
        "timestamp": "2024-04-18T20:57:22.446445773"
    }
}